runtime/src/snapshot_utils.rs

```diff
@@ -5459,4 +5513,65 @@ mod tests {
         let next_id = bank.accounts().accounts_db.next_id.load(Ordering::Relaxed) as usize;
         assert_eq!(max_id, next_id - 1);
     }
+
+    #[test]
+    fn test_bank_from_latest_snapshot_dir() {
+        solana_logger::setup();
+        let genesis_config = GenesisConfig::default();
+        let mut bank = Arc::new(Bank::new_for_tests(&genesis_config));
+
+        let tmp_dir = tempfile::TempDir::new().unwrap();
+        let bank_snapshots_dir = tmp_dir.path();
+        let collecter_id = Pubkey::new_unique();
+        let snapshot_version = SnapshotVersion::default();
+
+        for _ in 0..3 {
+            // prepare the bank
+            bank = Arc::new(Bank::new_from_parent(&bank, &collecter_id, bank.slot() + 1));
+            bank.fill_bank_with_ticks_for_tests();
+            bank.squash();
+            bank.force_flush_accounts_cache();
+
+            // generate the bank snapshot directory for slot+1
+            let snapshot_storages = bank.get_snapshot_storages(None);
+            let slot_deltas = bank.status_cache.read().unwrap().root_slot_deltas();
+            add_bank_snapshot(
+                bank_snapshots_dir,
+                &bank,
+                &snapshot_storages,
+                snapshot_version,
+                slot_deltas,
+            )
+            .unwrap();
+        }
+
+        let account_paths = &bank.rc.accounts.accounts_db.paths;
+        // Clear the contents of the account paths run directories.  When constructing the bank, the appendvec
+        // files will be extracted from the snapshot hardlink directories into these run/ directories.
+        for path in account_paths {
```

Comment:
if this is a necessary step for `bank_from_latest_snapshot_dir`, would it make sense for this behavior to live inside that function?

Reply:
The snapshot dir contains the full state including the paths of the appendvecs.  When constructing a bank from a snapshot dir, it restores the appendvec files to the <account_path>/run/ directories, in the same layout as when the snapshot is taken.  So it requires a clean set of <account_path> directories same as the snapshot time.

Yes, you are right, it should go into bank_from_latest_snapshot_dir.  Let me change that.

The PRE-state snapshot dirs should be removed.  I will also do this inside the function.

Reply:
I moved it to bank_from_snapshot_dir, and removed such code from the two test functions.